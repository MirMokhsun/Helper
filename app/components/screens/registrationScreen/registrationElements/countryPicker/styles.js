import { StyleSheet } from 'react-native';

export const styles = StyleSheet.create({
    wrapper: {
        width: 290,
        height: 40,
        padding: 0,
        backgroundColor: '#FFFFFF',
        justifyContent: 'center',
        flexDirection: 'row',
        justifyContent: 'space-between',
        alignItems: 'center',
        paddingRight: 15,
        borderBottomWidth: 1,
        borderColor: "#1459D2",
        paddingRight: 5,
        marginBottom: 10,
    },
    activetext: {
        fontSize: 18,
        fontFamily: 'Roboto-Light',
        color: '#1459D2',
        lineHeight: 21,
        textAlign: 'left',
    },
    svgWrapper: {
        width: 55,
        height: 40,
        flexDirection: 'row',
        justifyContent: 'space-between',
        alignItems: 'center',
        paddingRight: 5,
    },
    textWrapper: {
        width: 235,
        height: 40,
        justifyContent: 'center',
    }
});