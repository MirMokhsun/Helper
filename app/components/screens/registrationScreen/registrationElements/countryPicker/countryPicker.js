import React, { PureComponent } from 'react';
import { View, Text, TouchableOpacity } from 'react-native';
import { styles } from './styles';
import { withNavigation } from 'react-navigation';
import { flags } from '../../../countryScreen/countryScreenElements/png_flags';
import Polygon from '../../../../commonComponents/svg/Polygon';
import { setCountry } from '../../../../../api/setCountryName/setCountryName';

export class CountryPicker extends PureComponent {
    constructor(props) {
        super(props)
        this.state = {
            countryName: 'IO',
        }
    }

    componentDidMount() {
        const { countryKey, language } = this.props;
        this.setState({ countryName: setCountry(countryKey, language) });
    }

    UNSAFE_componentWillReceiveProps({ countryKey, language }) {
        const countryName = setCountry(countryKey, language);
        this.setState({ countryName })
    }

    navigateCountryScreen = () => {
        const { navigation } = this.props;
        navigation.navigate('CountryScreen');
    }

    render() {
        const { countryKey } = this.props;
        const { countryName } = this.state;
        return (
            <TouchableOpacity onPress={this.navigateCountryScreen} testID="BtnCountryPickerRegistrationID" accessibilityLabel="BtnCountryPickerRegistrationID">
                <View style={styles.wrapper}>
                    <View style={styles.textWrapper}>
                        <Text style={styles.activetext} numberOfLines={1}>{countryName}</Text>
                    </View>
                    <View style={styles.svgWrapper}>
                        {flags[countryKey]}
                        <Polygon />
                    </View>
                </View>
            </TouchableOpacity >
        )
    }
}

export default withNavigation(CountryPicker);