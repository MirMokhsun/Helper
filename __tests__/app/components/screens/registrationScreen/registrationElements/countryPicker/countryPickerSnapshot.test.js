import 'react-native';
import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import { CountryPicker } from '../../../../../../../app/components/screens/registrationScreen/registrationElements/countryPicker/countryPicker';

it('create snapshot CountryPicker', () => {
    const renderer = new ShallowRenderer()
    let toolbarSnapshot = renderer.render(<CountryPicker
        value={'value'}
        isValid={false}
        navigation={() => { }}
    />);
    expect(toolbarSnapshot).toMatchSnapshot();
});