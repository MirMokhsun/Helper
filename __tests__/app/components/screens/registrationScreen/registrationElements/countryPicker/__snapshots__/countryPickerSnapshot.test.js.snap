// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create snapshot CountryPicker 1`] = `
<TouchableOpacity
  accessibilityLabel="BtnCountryPickerRegistrationID"
  activeOpacity={0.2}
  onPress={[Function]}
  testID="BtnCountryPickerRegistrationID"
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "borderBottomWidth": 1,
        "borderColor": "#1459D2",
        "flexDirection": "row",
        "height": 40,
        "justifyContent": "space-between",
        "marginBottom": 10,
        "padding": 0,
        "paddingRight": 5,
        "width": 290,
      }
    }
  >
    <View
      style={
        Object {
          "height": 40,
          "justifyContent": "center",
          "width": 235,
        }
      }
    >
      <Text
        numberOfLines={1}
        style={
          Object {
            "color": "#1459D2",
            "fontFamily": "Roboto-Light",
            "fontSize": 18,
            "lineHeight": 21,
            "textAlign": "left",
          }
        }
      >
        IO
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "height": 40,
          "justifyContent": "space-between",
          "paddingRight": 5,
          "width": 55,
        }
      }
    >
      <Polygon />
    </View>
  </View>
</TouchableOpacity>
`;
